presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -6830,6 +6830,31 @@ public void testAlterColumnNotNull()
         assertUpdate(getSession(), dropTableStmt);
     }
 
+    @Test
+    public void testNoneAutoCommitTransactionWithFailAndRollback()
+    {
+        Session session = getQueryRunner().getDefaultSession();
+        assertUpdate(session, "CREATE TABLE test_non_autocommit_table (a bigint, b double, c varchar) with(partitioned_by = array['c'])");
+        transaction(getQueryRunner().getTransactionManager(), getQueryRunner().getAccessControl())
+                .execute(session, txnSession3 -> {
+                    assertUpdate(txnSession3, "insert into test_non_autocommit_table values(1, 1.1, '1001'), (2, 1.2, '1002')", 2);
+                    assertQueryFails(txnSession3, "insert into test_non_autocommit_table values(3, 1.3, '1002')",
```

Comment:
Similar to other comment, if you need to force a query failure I would use the `fail` function